### ISSUE TITLE
Caddyfile: redefining a site block to catch all logs does not use the new log definition

### ISSUE THREAD

**Issue by freydrich** [caddyserver/caddy#3507]
In Caddy v2.0.0, using the following Caddyfile example config, the expectation was that requests going to anything but http://192.168.1.1/folder1/* would be logged in the file access_default.log. 

However, although the response received is the expected status 501 and an empty body content, the access logs are still going into the access_folder1.log.

```
http://192.168.1.1/folder1/* {
        log {
                output file /var/log/caddy/access_folder1.log {
                        roll_size 100mb
                        roll_keep 2
                        roll_keep_for 48h
                }
        }

        reverse_proxy /folder1/* 127.0.0.1:8080 {
        }
}

http:// {
        log {
                output file /var/log/caddy/access_default.log {
                        roll_size 100mb
                        roll_keep 10
                        roll_keep_for 2160h
                }
        }

        respond 501 {
                body ""
                close
        }
}
```

Here is a request example that would reproduce the wrong behavior:
`curl http://192.168.1.1/`
The access output is logged to access_folder1.log instead of access_default.log

**Comment by mholt** [comment]
Yeah, `log` directives aren't handlers, so they can't be scoped by request path. They're a server-based config parameter: https://caddyserver.com/docs/json/apps/http/servers/logs/

This is one of the unfortunate limitations of the Caddyfile.

Also, Caddy's logging philosophy is "emit anything and everything, process them later": https://caddyserver.com/docs/logging - it's much simpler, and totally workable because logs are structured unlike old-school logs that don't even show Host information on the requests.

**Comment by freydrich** [comment]
Thanks for the explanation, I guess it makes sense.

However this is actually what I was trying to do initially: to define the log globally (but using the  Caddyfile) for all my sites, however I did not succeeed with v2 at first, so I understood the only way was to put the log statement in each site block (which I thought didn't make sense...).

So I started doing this in each site block and I thought about logging this way (in different files) at the same time so that I can catch unpexpected requests on my server easily without having to filter or process logs further.

I may just need to find a specific example with a Caddyfile on how to define log only once for all sites. I'll do a bit more reading to see if I can figure out a better approach.
